htdocs/core/lib/functions.lib.php

```diff
@@ -10831,35 +10832,101 @@ function getElementProperties($element_type)
 		'classfile' => $classfile,
 		'classname' => $classname
 	);
+
+
+
+	if (!class_exists('ElementProperties')) { require_once DOL_DOCUMENT_ROOT . '/core/class/elementproperties.class.php'; }
```

Comment:
This does not go in the good direction.
We try to have generic code working with "code convention". Introducing sql for a such often used code will allow developer to break code convention.
We should be able to have something working completely with code convention.